haskell-src/Concordium/Types/Transactions.hs

```diff
@@ -619,6 +777,26 @@ instance TransactionData Transaction where
     transactionSignHash = atrSignHash . wmdData
     transactionHash = wmdHash
 
+instance TransactionData AccountTransactionV1 where
```

Comment:
I would extend `TransactionData` with `transactionSponsor :: t -> Maybe AccountAddress` and `transactionSponsorSignature :: t -> Maybe TransactionSignature`. They should both return `Nothing` for old-style transactions.